full-service/src/service/sync.rs

```diff
@@ -143,141 +142,132 @@ pub fn sync_account(
     account_id_hex: &str,
     logger: &Logger,
 ) -> Result<(), SyncError> {
-    let conn = wallet_db.get_conn()?;
-
     while let SyncStatus::ChunkFinished =
-        sync_account_next_chunk(ledger_db, &conn, logger, account_id_hex)?
+        sync_account_next_chunk(ledger_db, wallet_db, logger, account_id_hex)?
     {}
 
     Ok(())
 }
 
 fn sync_account_next_chunk(
     ledger_db: &LedgerDB,
-    conn: &Conn,
```

Comment:
This is because the function to decode a subaddress index now requires a database connection, which is not thread safe, so in order to maintain the `into_par_iter().filter_map(..)` when decoding the individual received txos, I pass the wallet_db, which is thread safe, and pool for the connection when required

Reply:
Was necessary until a refactor, so it can probably just go back so we don't worry about connection failures during syncing a chunk

Reply:
I believe the biggest advantage for using the same connection during sync is that it allows you to execute the entire block sync in a single SQL transaction, so you can't end up with a partially synced block if something fails midway.

Reply:
right, okay, lemme fix the transaction chunk, too